src/lambda/commands/importLambda.ts

```diff
@@ -0,0 +1,233 @@
+import { makeTemporaryToolkitFolder, fileExists } from '../../shared/filesystemUtilities'
+import { getLogger } from '../../shared/logger'
+import { createCodeAwsSamDebugConfig } from '../../shared/sam/debugger/awsSamDebugConfiguration'
+import { ExtensionDisposableFiles } from '../../shared/utilities/disposableFiles'
+import * as pathutils from '../../shared/utilities/pathUtils'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+import { Window } from '../../shared/vscode/window'
+import { LambdaFunctionNode } from '../explorer/lambdaFunctionNode'
+import { getFamily, RuntimeFamily } from '../models/samLambdaRuntime'
+import { showConfirmationMessage } from '../../s3/util/messages'
+import { addWorkspaceFolder } from '../../shared/utilities/workspaceUtils'
+
+const pipeline = promisify(Stream.pipeline)
+
+export async function importLambdaCommand(functionNode: LambdaFunctionNode, window = Window.vscode()) {
+    const workspaceFolders = vscode.workspace.workspaceFolders || []
+    const labelToWorkspace = new Map(workspaceFolders.map(folder => [`$(root-folder-opened) ${folder.name}`, folder]))
+    const otherLocationName = `$(folder-opened) ${localize('AWS.lambda.import.otherLocation', 'Select a folder...')}`
+
+    // TODO: Move to FolderQuickPickItem (would have to modify BrowseFolderQuickPickItem to non-Wizard dependence)?
+    const selectedLocation = await vscode.window.showQuickPick([...labelToWorkspace.keys(), otherLocationName], {
+        placeHolder: localize('AWS.lambda.import.prompt.placeholder', 'Select the import location'),
+    })
+
+    if (!selectedLocation) {
+        getLogger().info('ImportLambda cancelled')
+        return
+    }
+
+    let selectedUri: vscode.Uri
+    const isWorkspaceFolder = selectedLocation != otherLocationName
+    if (isWorkspaceFolder) {
+        selectedUri = labelToWorkspace.get(selectedLocation)!.uri
+    } else {
+        const selectedDirectory = await window.showOpenDialog({
+            openLabel: 'Select',
+            canSelectFiles: false,
+            canSelectFolders: true,
+        })
+        if (!selectedDirectory) {
+            getLogger().info('ImportLambda cancelled')
+            return
+        }
+        selectedUri = selectedDirectory[0]
+        addWorkspaceFolder({ uri: selectedUri })
+    }
+
+    const functionName = functionNode.configuration.FunctionName!
+
+    const importLocation = path.join(selectedUri.fsPath, functionName, path.sep)
+    const importLocationName = vscode.workspace.asRelativePath(importLocation, true)
+    const workspaceFolder = isWorkspaceFolder
+        ? labelToWorkspace.get(selectedLocation)!
+        : vscode.workspace.getWorkspaceFolder(vscode.Uri.file(importLocation))!
+
+    const directoryExists = await fs.pathExists(importLocation)
+
+    const overwriteWarning = localize(
+        'AWS.lambda.import.overwriteWarning',
+        '\nA directory named {0} already exists! Importing will overwrite any existing files!\n',
```

Comment:
Message could be made more concise and less exclamatory:

`{0} already exists. Existing files will be overwritten.`

Reply:
could also put the variable part at the end, which helps the reader (purpose of the message is always in the same visual location, instead of varying based on filename):

    Directory already exists and will be overwritten: {0}